drivers/ieee802154/ieee802154_stm32wba.c

```diff
@@ -0,0 +1,1079 @@
+		st_ral_set_max_frame_retries(config->max_frm_retries);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_MAX_CSMA_FRAME_RETRIES:
+		LOG_DBG("Setting MAX_CSMA_FRAME_RETRIES: %u", config->max_csma_frm_retries);
+		st_ral_set_max_csma_frame_retries(config->max_csma_frm_retries);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_MIN_CSMA_BE:
+		LOG_DBG("Setting MIN_CSMA_BE: %u", config->min_csma_be);
+		st_ral_set_min_csma_be(config->min_csma_be);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_MAX_CSMA_BE:
+		LOG_DBG("Setting MAX_CSMA_BE: %u", config->max_csma_be);
+		st_ral_set_max_csma_be(config->max_csma_be);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_MAX_CSMA_BACKOFF:
+		LOG_DBG("Setting MAX_CSMA_BACKOFF: %u", config->max_csma_backoff);
+		st_ral_set_max_csma_backoff(config->max_csma_backoff);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_IMPLICIT_BROADCAST:
+		LOG_DBG("Setting IMPLICIT_BROADCAST: %u", config->impl_brdcast);
+		st_ral_set_implicitbroadcast(config->impl_brdcast);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_ANTENNA_DIV:
+		LOG_DBG("Setting ANTENNA_DIV: %u", config->ant_div);
+		if (st_ral_set_ant_div_enable(config->ant_div)) {
+			return -EIO;
+		}
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_RADIO_RESET:
+		LOG_DBG("Setting RADIO_RESET");
+		if (st_ral_radio_reset()) {
+			return -EIO;
+		}
+		break;
+
+	default:
+		LOG_ERR("Unsupported configuration type: %d", type);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int stm32wba_attr_get_extended(enum ieee802154_stm32wba_attr attr,
+				      const struct ieee802154_stm32wba_attr_value *value)
+{
+	switch ((uint32_t)attr) {
+	case IEEE802154_STM32WBA_ATTR_CCA_THRESHOLD:
+		LOG_DBG("Getting CCA_THRESHOLD attribute");
+		static int8_t l_cca_thr;
+
+		if (st_ral_get_cca_energy_detect_threshold(&l_cca_thr)) {
+			return -ENOENT;
+		}
+		((struct ieee802154_stm32wba_attr_value *)value)->cca_thr = &l_cca_thr;
+		break;
+
+	case IEEE802154_STM32WBA_ATTR_IEEE_EUI64:
+		LOG_DBG("Getting IEEE_EUI64 attribute");
+		uint8_t l_eui64[8];
+
+		stm32wba_get_eui64(l_eui64);
+		memcpy(((struct ieee802154_stm32wba_attr_value *)value)->eui64, l_eui64,
+				sizeof(l_eui64));
+		break;
+
+	case IEEE802154_STM32WBA_ATTR_TX_POWER:
+		LOG_DBG("Getting TX_POWER attribute");
+		static uint8_t l_tx_power;
+
+		if (st_ral_tx_power_get(&l_tx_power)) {
+			return -ENOENT;
+		}
+		((struct ieee802154_stm32wba_attr_value *)value)->tx_power = &l_tx_power;
+		break;
+
+	case IEEE802154_STM32WBA_ATTR_RAND_NUM:
+		LOG_DBG("Getting RAND_NUM attribute");
+		static uint8_t l_rand_num;
+
+		if (st_ral_mac_gen_rnd_num(&l_rand_num, 1, true)) {
+			return -ENOENT;
+		}
+		((struct ieee802154_stm32wba_attr_value *)value)->rand_num = &l_rand_num;
+		break;
+
+	default:
+		LOG_ERR("Unsupported attribute: %u", (uint32_t)attr);
+		return -ENOENT;
+	}
+
+	return 0;
+}
+#endif
+
+/* Radio device API */
+
+static enum ieee802154_hw_caps stm32wba_get_capabilities(const struct device *dev)
+{
+	return stm32wba_data.capabilities;
+}
```

Comment:
```suggestion
static enum ieee802154_hw_caps stm32wba_get_capabilities(const struct device *dev)
{
	struct stm32wba_802154_data *data = STM32WBA_802154_DATA(dev);

	return data->capabilities;
}
```

(or remain single instance, but in this case the whole `stm32wba_data.capabilities` should be removed - just return the value computed by `stm32wba_get_capabilities_at_boot` here, since that's the only place where it is used(?))

Reply:
stm32wba_get_capabilities() directly returns mask value previously present in stm32wba_get_capabilities_at_boot().
capabilities field is removed in the stm32wba_802154_data structure